uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/toil-request.controller.js

```diff
@@ -102,29 +101,26 @@ define([
      * @return {Boolean}
      */
     function checkSubmitConditions () {
-      return !!vm.request.from_date && !!vm.request.to_date &&
-        !!vm.request.toil_duration && !!vm.request.toil_to_accrue;
+      return !!parentCtrl.request.from_date && !!parentCtrl.request.to_date &&
+        !!parentCtrl.request.toil_duration && !!parentCtrl.request.toil_to_accrue;
     }
 
     /**
      * Extends parent method. Fires calculation of expiry date when the
      * number of days changes and the expiry date can be calculated.
+     *
+     * @return {Promise}
      */
-    function changeInNoOfDays () {
-      vm._reset();
-      vm._calculateOpeningAndClosingBalance();
-
-      if (canCalculateExpiryDate()) {
-        vm.calculateToilExpiryDate();
-      }
+    function changeInNoOfDaysExtended () {
```

Comment:
Don't call it `extended`, it's just confusing. Honestly I can't think of a better name than a `onXYZ` one: `onChangeInNoOfDays`, because really this is a generic event handler that a child controller can have

Reply:
applies to `setDatesFromUIExtended` also

Reply:
I don't agree with `onChangeInNoOfDays` since this could easily be confused with `changeInNoOfDays`. I could not come up with a better name either that's why I chose this.

Reply:
I attempted to improve that:

```js
/**
 * Sets day selection mode: multiple days or a single day
 *
 * @return {Promise}
 */
function setDaysSelectionMode () {
  if ((!vm.isMode('create') && requestHasSameDates()) ||
    (vm.isMode('create') && (isLeaveType('sickness') || isCalculationUnit('hours')))) {
    vm.uiOptions.multipleDays = false;
  } else {
    vm.uiOptions.multipleDays = true;
  }

  return $q.resolve().then(vm.setDaysSelectionModeExtended);
}
```

Called on init and on period absence type change.

```js
/**
 * Handles the change of days selection mode (single day, multiple days)
 *
 * @return {Promise}
 */
function daysSelectionModeChangeHandler () {
  return attemptCalculateBalanceChange()
    .then(vm.setDaysSelectionModeExtended);
}
```

bound to vm, called from UI only.